src/App.tsx

```diff
@@ -15,48 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+type Reset = (
+  handleSorted: (sorted: SortType) => void,
+  handleReversed: (isRevers: boolean) => void,
+) => void;
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default: return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsRevesed] = useState(false);
+  const visibleGoods = getReorderedGoods(
+    goodsFromServer, { sortType, isReversed },
+  );
+
+  const handleSort = (typeSort: SortType) => setSortType(typeSort);
+  const handleReverse = (isRevers: boolean) => setIsRevesed(isRevers);
+  const reset: Reset = (handleSorted, handleReversed) => {
+    handleSorted(SortType.NONE);
+    handleReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button is-info',
+            { 'is-light': sortType !== 1 })}
+          onClick={() => handleSort(SortType.ALPHABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames('button is-success',
+            { 'is-light': sortType !== 2 })}
+          onClick={() => handleSort(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames('button is-warning',
+            { 'is-light': !isReversed })}
+          onClick={() => handleReverse(!isReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortType !== 0 || isReversed) && (
```

Comment:
- use enum
- I'd rather put it to another variable

Reply:
Done